### PR TITLE
added routes for meals and reservation

### PR DESCRIPTION
**Pull request by bhumikama**
[This PR includes includes endpoints for meal sharing](https://github.com/HackYourFuture-CPH/node.js/blob/main/week2/homework/readme.md)

 two categories of routes are added: meals and reservations.
/api/src/routers/meals.js => Meals Router
/api/src/routers/reservations.js => Reservation Router

### CODE REVIEW

File: api/src/routers/meals.js [hunk 1]
```diff
@@ -0,0 +1,75 @@
+import express from "express";
+import knex from "../database_client.js";
+
+const mealsRouter = express.Router();
+
+//return all meals
+mealsRouter.get("/", async (req, res, next) => {
+  try {
+    const meals = await knex("Meal").select("*");
```
Comment (urbanogilson) [inline api/src/routers/meals.js]: Select is implicit

```suggestion
    const meals = await knex("Meal");
```

File: api/src/routers/meals.js [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    next(error);
+  }
+});
+
+//insert meal
+mealsRouter.post("/", async (req, res, next) => {
+  try {
+    console.log(req.body);
+    const data = req.body;
+    await knex("Meal").insert(data);
```
Comment (urbanogilson) [inline api/src/routers/meals.js:21]: You named your table `Meal`, checkout this page for SQL table/column name conventions: [https://brainstation.io/learn/sql/naming-conventions](https://brainstation.io/learn/sql/naming-conventions)

File: api/src/routers/meals.js [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+  } catch (error) {
+    next(error);
+  }
+});
+
+//GET meals by id
+mealsRouter.get("/:id", async (req, res, next) => {
+  try {
+    const id = req.params.id;
+    const meal = await knex("Meal").select("*").where("id", id);
```
Comment (urbanogilson) [inline api/src/routers/meals.js]: We expect an object but `where` returns a list. Also, it's returning an empty list when the meal is not found, not 404.

Add `first` at the end to return one object.

```suggestion
    const meal = await knex("Meal").where("id", id).first();
```

File: api/src/routers/meals.js [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+      res.status(200).json({ message: "deleted successfully" });
+    } else {
+      res.status(404).json({ message: "Meal not found" });
+    }
+  } catch (error) {
+    next(error);
+  }
+});
+
+export default mealsRouter;
```
Comment (urbanogilson) [inline api/src/routers/meals.js:75]: The routes are not enabled in the `index.js`

File: api/src/routers/reservations.js [hunk 5]
```diff
@@ -0,0 +1,77 @@
+import express from "express";
+import knex from "../database_client.js";
+
+const reservationRouter = express.Router();
+
+//Returns all reservations
+reservationRouter.get("/", async (req, res, next) => {
+    try {
+        const reservation = await knex("Reservation").select("*");
```
Comment (urbanogilson) [inline api/src/routers/reservations.js]:
```suggestion
        const reservation = await knex("Reservation")
```

File: api/src/routers/reservations.js [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+    } catch (error) {
+        next(error);
+    }
+});
+
+//GET reservations by id
+reservationRouter.get("/:id", async (req, res, next) => {
+    try {
+        const id = req.params.id;
+        const reservation = await knex("Reservation").select('*').where('id',id);
```
Comment (urbanogilson) [inline api/src/routers/reservations.js]:
```suggestion
        const reservation = await knex("Reservation").where('id',id).first();
```

File: api/src/routers/reservations.js [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+        else{
+            res.status(404).json({ "message": "Reservation not found" });
+        }
+        
+    } catch (error) {
+        next(error);
+    }
+});
+
+export default reservationRouter;
```
Comment (urbanogilson) [inline api/src/routers/reservations.js:77]: The routes are not enabled in the index.js

File: api/src/routers/meals.js [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+    res.json(meals);
+  } catch (error) {
+    next(error);
+  }
+});
+
+//insert meal
+mealsRouter.post("/", async (req, res, next) => {
+  try {
+    console.log(req.body);
```
Comment (urbanogilson) [inline api/src/routers/meals.js:19]:
```suggestion
```

File: api/src/routers/reservations.js [hunk 9]
```diff
@@ -0,0 +1,77 @@
...
+        res.json(reservation);
+    } catch (error) {
+        next(error);
+    }
+});
+
+//Adds a new reservation to the database
+reservationRouter.post("/", async (req, res, next) => {
+    try {
+        console.log(req.body);
```
Comment (urbanogilson) [inline api/src/routers/reservations.js:19]:
```suggestion
```

File: api/src/routers/reservations.js [hunk 10]
```diff
@@ -0,0 +1,77 @@
...
+
+//Deletes the Reservation by id
+reservationRouter.delete("/:id", async (req, res, next) => {
+    try {
+        const id = req.params.id;
+        const deletedReservation = await knex("Reservation").where('id',id).del();
+        if(deletedReservation){
+            res.status(200).json({ "message": "deleted successfully" });
+        }
+        else{
```
Comment (urbanogilson) [inline api/src/routers/reservations.js:68]: Use Prettier

```suggestion
        else {
```

**Review by urbanogilson — COMMENTED** [review]
Good job! Left inline comments

**Comment by bhumikama** [comment]
Thanks @urbanogilson  for your review.
I have made changes according to your inputs. Please have another look.

**Pull request merged**